lib/internal/repl.js

```diff
@@ -84,17 +99,22 @@ function setupHistory(repl, historyPath, ready) {
     if (err) {
       return ready(err);
     }
-
     if (data) {
+      repl.history = data.split(/\n|\r/).slice(-repl.historySize);
```

Comment:
FWIW the regex I suggested has an interesting side-effect: empty lines will be ignored.